proto/spire/common/common.proto

```diff
@@ -125,9 +125,6 @@ message RegistrationEntries {
 /** Certificate represents a ASN.1/DER encoded X509 certificate */
 message Certificate {
     bytes der_bytes = 1;
-
-    /** Whether or not the key used to sign the certificate is tainted */
-    bool tainted_key = 2;
```

Comment:
Why move this to the top level? Feels natural to have on a per-cert and per-key basis ... it remains set on PublicKey message

Reply:
the issue here is that we does not have intermediate certificates on bundle when UpstreamAuthority is used, 
So we keep "only" bundle here in that case, but intermediates lives on Journal,
So in order to taint a key in that case we must store the public key in some place